### PR TITLE
VelocityControl python bindings

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

Pushing `VelocityControl` struct into python bindings.

Supporting changes:
 - added get/set for gravity to Simulator bindings
 - `VelocityControl` now stored and used via smart pointers
 - python examples' `default_sim_settings` `'test_scene'` field renamed to `'scene'`. Previously, the `default_sim_settings` was not a valid configuration without specifying a scene.
 - updating magnum submodule for a `math::angle` patch.

## How Has This Been Tested

New python test `tests/test_physics::test_velocity_control` and changes to `tests/test_physics::test_dynamics` for gravity.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: tests/test_physics.py [hunk 1]
```diff
@@ -189,5 +195,96 @@ def test_dynamics(sim):
...
+
+        # NOTE: explicit integration, so expect some error
+        ground_truth_q = mn.Quaternion([[1.0, 0, 0], 0])
+        assert np.allclose(
+            sim.get_translation(object_id), np.array([0, 1.0, 0.0]), atol=0.03
+        )
+        angle_error = mn.math.angle(ground_truth_q, sim.get_rotation(object_id))
+        if not math.isnan(angle_error):
+            # nan indicates very low angle error
+            assert angle_error < mn.Rad(0.05)
```
Comment (mosra) [inline tests/test_physics.py:288]: You mentioned this on Slack -- I'll try to fix today so you can remove this piece of code before it's merged in.

File: tests/test_physics.py [hunk 2]
```diff
@@ -189,5 +195,96 @@ def test_dynamics(sim):
...
+    cfg_settings = examples.settings.default_sim_settings.copy()
+    cfg_settings["scene"] = "NONE"
+    cfg_settings["enable_physics"] = True
+    hab_cfg = examples.settings.make_cfg(cfg_settings)
+    sim.reconfigure(hab_cfg)
+    sim.set_gravity(np.array([0, 0, 0.0]))
+
+    sphere_path = osp.abspath("data/test_assets/objects/sphere")
+    template_ids = sim.load_object_configs(sphere_path)
+    object_template = sim.get_object_template(template_ids[0])
```
Comment (pushkalkatara) [inline tests/test_physics.py:224]: Do I need to load template if I use `MotionType.KINEMATIC`?

I am trying to use velocity control with the agent.

**Comment by pushkalkatara** [comment]
```python
def test_agent_forward():
    agent_node = sim.agents[0].scene_node
    sim.add_object(0, agent_node)
    #sim.set_translation(np.random.rand(3), object_id)
    #assert np.allclose(agent_node.translation, sim.get_translation(object_id))

    vel_control = sim.get_object_velocity_control(object_id)
    # test global velocities
    vel_control.linear_velocity = np.array([1.0, 0, 0])
    vel_control.angular_velocity = np.array([0, 1.0, 0])
    vel_control.controlling_lin_vel = True
    vel_control.controlling_ang_vel = True

    old_translation = sim.get_translation(object_id)
    while sim.get_world_time() < 1.0:
        # NOTE: stepping close to default timestep to get near-constant velocity control of DYNAMIC bodies.
        sim.step_physics(0.00416)

    ground_truth_pos = sim.get_world_time() * vel_control.linear_velocity + old_translation
    print(sim.get_translation(object_id))
    print(ground_truth_pos)
    assert np.allclose(sim.get_translation(object_id), ground_truth_pos, atol=0.01)
```

The test fails for translating agent. The output is
```
agent translation after applying linear vel: Vector(-0.436783, 0.107289, 8.58296)
ground truth pose: Vector(-0.369022, 0.0843179, 8.60693)
```

There are changes in Y and Z as well applying X velocities only.

File: tests/test_physics.py (hunk 1 above)
Comment (mosra) [reply tests/test_physics.py:288]: Fixed in mosra/magnum@498f4c59e1186a46cd8b30547d9383a68498a8a5 -- feel free to update the submodule here.

**Comment by aclegg3** [comment]
> Is magnum being updated/changed on purpose?

Yes, we are grabbing a patch to prevent `math::angle(q0, q1)` from returning nan when the quaternions are very close.

**Comment by mosra** [comment]
There's a few new features pulled in with the submodule update (mainly [shader instancing](https://doc.magnum.graphics/magnum/classMagnum_1_1Shaders_1_1Phong.html#Shaders-Phong-instancing)), which is why it looks so big, but nothing that could cause regressions.

**Pull request merged**